test/javascript/components/concept-map/ConceptMap.test.tsx

```diff
@@ -8,78 +8,82 @@ import '@testing-library/jest-dom/extend-expect'
 
 // Component
 import { ConceptMap } from '../../../../app/javascript/components/concept-map/ConceptMap'
+import { IConceptMap } from '../../../../app/javascript/components/concept-map/concept-map-types'
 
 describe('<ConceptMap />', () => {
-  test('renders empty component', () => {
-    const { container } = render(
-      <ConceptMap
-        concepts={[]}
-        levels={[[]]}
-        connections={[]}
-        status={{}}
-        exercise_counts={{}}
-      />
-    )
+  test('renders empty component', async () => {
+    const config: IConceptMap = {
+      concepts: [],
+      levels: [[]],
+      connections: [],
+      status: {},
+      exercise_counts: {},
+    }
+
+    const { container } = await waitForConceptMapReady(config)
     const map = container.querySelector('.c-concepts-map')
     expect(map).not.toBeNull()
   })
 
   test('renders single incomplete concept', async () => {
     const testConcept = concept('test')
 
-    render(
-      <ConceptMap
-        concepts={[testConcept]}
-        levels={[[testConcept.slug]]}
-        connections={[]}
-        status={{
-          test: 'unavailable',
-        }}
-        exercise_counts={{
-          test: {
-            exercises: 1,
-            exercises_completed: 0,
-          },
-        }}
-      />
-    )
-
-    await waitFor(() => {
-      expect(screen.getByText('Test')).toBeInTheDocument()
-      expect(
-        screen.queryByTitle('You have mastered this concept')
-      ).not.toBeInTheDocument()
-    })
+    const config: IConceptMap = {
+      concepts: [testConcept],
+      levels: [[testConcept.slug]],
+      connections: [],
+      status: { test: 'unavailable' },
+      exercise_counts: { test: { exercises: 1, exercises_completed: 0 } },
+    }
+
+    await waitForConceptMapReady(config)
+
+    expect(
```

Comment:
You probably want a positive expectation here too. (e.g. expect that the concept is visible) -- even if `waitForConceptMapReady` already checks that. That will allow you to _change_ the control logic, but keep the test coherent/the same.